src/codegen/generators/expression_generator.rs

```diff
@@ -204,13 +204,7 @@ impl<'a, 'b> ExpressionCodeGenerator<'a, 'b> {
                         self.generate_expression(right)?,
                     ))
                 } else {
-                    Err(Diagnostic::codegen_error(
-                        &format!(
-                            "invalid types, cannot generate binary expression for {:?}",
-                            self.get_type_hint_info_for(expression)?
-                        ),
-                        left.get_location(),
-                    ))
+                    self.create_llvm_generic_binary_expression(operator, left, right, expression)
```

Comment:
How are Subranges treated here `INT(50)`?
I know it's not string related, but I wonder if we do a normal comparison?
Maybe ltype/rtype above should resolve to intrinsic types for this operation?
Do you also mind adding a few Subrange tests?

Reply:
the subranges should have been handled by the if's above (is_int() or is_float())

but i can add a test - no problemo

Reply:
👌 good review
ranges where a problem - i forgot that `find_effective_type` returns the range, not the referenced type behind it.
I introduced a `get_intrinsic_type`